Supplemental/MakeAverageDataset/MakeAverageDataset.sh

```diff
@@ -32,1086 +32,747 @@
 
-	if [ -z "${p_RegName}" ]; then
-		log_Err "reg name (--reg-name=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "reg name: ${p_RegName}"
-	fi
+opts_AddMandatory '--greyscale-maps' 'GreyScaleMaps' 'mapstring@mapstring' "Maps you want to use the grayscale palette (@ delimited list) (e.g. sulc@curvature)"
 
-	if [ -z "${p_VideenMaps}" ]; then
-		log_Err "videen maps (--videen-maps=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "videen maps: ${p_VideenMaps}"
-	fi
+opts_AddMandatory '--distortion-maps' 'DistortionMaps' 'mapstring@mapstring' "Distortion maps (@ delimited list) (e.g. SphericalDistortion@ArealDistortion@EdgeDistortion)"
 
-	if [ -z "${p_GreyScaleMaps}" ]; then
-		log_Err "greyscale maps (--greyscale-maps=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "greyscale maps: ${p_GreyScaleMaps}"
-	fi
+opts_AddMandatory '--gradient-maps' 'GradientMaps' 'mapstring@mapstring' "Maps you want to compute the gradietn on (@ delimited list) (e.g. MyelinMap_BC@SmoothedMyelinMap_BC@corrThickness)"
 
-	if [ -z "${p_DistortionMaps}" ]; then
-		log_Err "distortion maps (--distortion-maps=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "distortion maps: ${p_DistortionMaps}"
-	fi
+opts_AddMandatory '--std-maps' 'STDMaps' 'mapstring@mapstring' "maps you want to compute a standard deviation on (@ delimited list) (e.g. sulc@curvature@corrThickness@thickness@MyelinMap_BC)"
 
-	if [ -z "${p_GradientMaps}" ]; then
-		log_Err "gradient maps (--gradient-maps=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "gradient maps: ${p_GradientMaps}"
-	fi
+opts_AddMandatory '--multi-maps' 'MultiMaps' 'mapstring@mapstring' "Maps with more than one map (column) that cannot be merged and must be averaged (@ delimited list) (e.g. NONE)"
 
-	if [ -z "${p_STDMaps}" ]; then
-		log_Err "std maps (--std-maps=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "std maps: ${p_STDMaps}"
-	fi
 
-	if [ -z "${p_MultiMaps}" ]; then
-		log_Err "multi maps (--multi-maps=) required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "multi maps: ${p_MultiMaps}"
-	fi
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
 
-	if [ -z "${p_NoMergedT1T2vols}" ]; then
-		log_Err "No Merged T1T2 Volumes value required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "No Merged T1T2 Volumes: ${p_NoMergedT1T2vols}"
-	fi
+#display the parsed/default values
+opts_ShowValues
 
-	if [ -z "{p_NoLabelVols}" ]; then
-		log_Err "No Label Volumes value required"
-		error_count=$(( error_count + 1 ))
-	else
-		log_Msg "No Label Volumes: ${p_NoLabelVols}"
-	fi
+log_Check_Env_Var HCPPIPEDIR
+log_Check_Env_Var CARET7DIR
+log_Check_Env_Var FSLDIR
 
-	if [ ${error_count} -gt 0 ]; then
-		log_Err_Abort "For usage information, use --help"
-	fi
-}
+
+local error_count=0
```

Comment:
`error_count` is part of the old parsing logic, and should be removed.

Reply:
Done